server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -850,6 +857,11 @@ public void initRestHandlers(Supplier<DiscoveryNodes> nodesInCluster) {
             }
         }
         registerHandler.accept(new RestCatAction(catActions));
+
+        // Remote Store APIs
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)) {
```

Comment:
@andrross Added check for feature flag before registering the rest endpoint.

If feature flag is disabled, we get following output:
```
$ curl -X POST "localhost:9200/_remotestore/_restore" -H 'Content-Type: application/json' -d'{"indices": "my-index-1"}'

{"error":"no handler found for uri [/_remotestore/_restore] and method [POST]"}%
```